components/brave_wallet_ui/components/extension/connect-with-site-panel/connect-with-site-header/connect-with-site-header.style.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as leo from '@brave/leo/tokens/css'
+import Icon from '@brave/leo/react/icon'
+import styled from 'styled-components'
+import { Row, WalletButton } from '../../../shared/style'
+
+export const StyledWrapper = styled.div<{ isScrolled: boolean }>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  top: 48px;
+  left: 0;
+  right: 0;
+  position: fixed;
+  z-index: 8;
+  box-sizing: border-box;
+  opacity: ${(p) => (p.isScrolled ? 0 : 1)};
+  transition-duration: 0.4s;
+  transition-timing-function: ease-out;
+  transition-delay: 0s;
+`
+
+export const TitleWrapper = styled(Row)<{
+  isReadyToConnect: boolean
+  isScrolled: boolean
+}>`
+  position: relative;
+  z-index: 20;
+  background-color: ${leo.color.container.background};
+  box-shadow: ${(p) =>
+    p.isScrolled ? '0px 3px 10px 1px rgba(0, 0, 0, 0.08)' : 'none'};
```

Comment:
We should follow up to polish this transition as well

Reply:
Ditto on issue created: https://github.com/brave/brave-browser/issues/29299